.github/workflows/cabal-install.yml

```diff
@@ -36,6 +36,7 @@ jobs:
         ghc-version: '9.10'
     - name: Configure the build plan
       run: |
+        # shellcheck disable=SC2086
```

Comment:
This incantation looks very mysterious to people who aren't high-level bash wizards.

Not sure where the best place is to explain it though, since these appear in a lot of places. Possibly add some notes about the ci setup to `HACKING.md` under "Testing with Github Actions".

Reply:
I'll take this in another PR.